src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs

```diff
@@ -452,22 +452,25 @@ public StringSegment Subsegment(int offset, int length)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public int IndexOf(char c, int start, int count)
         {
-            int offset = Offset + start;
+            int index = -1;
 
-            if (!HasValue || start < 0 || (uint)offset > (uint)Buffer.Length)
+            if (HasValue)
             {
-                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.start);
-            }
+                if (start < 0 || start > Length)
```

Comment:
```suggestion
                if ((uint)start > (uint)Length)
```
to avoid one comparison. If negative the cast to `uint` will put into the range > int.MaxValue, so it's safe.